feat(ECO-2984): Add my favorites Move package (#675)
econia-labs/emojicoin-dot-fun: src/move/favorites/README.md

<!-- cspell:word permissioned -->

# `emojicoin-dot-fun` favorites

> This package contains functions to set, get and unset favorite markets per user.

Reviewer: ```suggestion
This package contains functions to add, remove, and view a user's favorite
emojicoin markets.
```

## Publish commands

Set variables:

```sh
FAVORITES=0xaaa...
EMOJICOIN_DOT_FUN=0xbbb...

> PROFILE=my-profile

Reviewer: that's true

```

Publish:

```sh
NAMED_ADDRESSES=$(
    printf "%s,%s" \

>         "favorites=$FAVORITES" \

Reviewer: ```suggestion
        "favorites=$PUBLISHER" \
```

the CLI automatically resolves profiles to their corresponding address when used like this

        "emojicoin_dot_fun=$EMOJICOIN_DOT_FUN"
)
aptos move publish \
    --assume-yes \
    --move-2 \
    --named-addresses $NAMED_ADDRESSES \
    --profile $PROFILE
```

> ## Set favorite

Reviewer: ```suggestion
## Add favorite
```

```sh
MARKET_ADDRESS=0xccc...
FAVORITES=0xaaa...
PROFILE_NAME=my-profile

aptos move run \
  --args address:$MARKET_ADDRESS \

>   --function-id $FAVORITES::emojicoin_dot_fun_favorites::set_favorite \

Reviewer: ```suggestion
  --function-id $FAVORITES::emojicoin_dot_fun_favorites::add_favorite \
```

  --profile $PROFILE_NAME
```

> ## Unset favorite

Reviewer: ```suggestion
## Remove favorite
```

```sh
MARKET_ADDRESS=0xccc...
FAVORITES=0xaaa...
PROFILE_NAME=my-profile

aptos move run \
  --args address:$MARKET_ADDRESS \

>   --function-id $FAVORITES::emojicoin_dot_fun_favorites::unset_favorite \

Reviewer: ```suggestion
  --function-id $FAVORITES::emojicoin_dot_fun_favorites::remove_favorite \
```

  --profile $PROFILE_NAME
```

## Get favorites

```sh
USER_ADDRESS=0xccc...
FAVORITES=0xaaa...
PROFILE_NAME=my-profile

aptos move view \
  --args address:$USER_ADDRESS \

>   --function-id $FAVORITES::emojicoin_dot_fun_favorites::favorites \

Reviewer: ```suggestion
  --function-id $FAVORITES::emojicoin_dot_fun_favorites::view_favorites \
```

> <!-- cspell:word permissioned -->
> 
> # `emojicoin-dot-fun` favorites

Reviewer 2: ```suggestion
# `emojicoin-dot-fun` favorites
```
